### DISCUSSION TITLE
Ability to build images on K3S

### DISCUSSION THREAD

**Discussion by egidijus** [k3s-io/k3s#7305] [General]
**Build images on K3S**

* I think I have rancher, ```sudo ps -aux| grep ranch
root        2146  3.5  0.4 775236 79084 ?        Sl   07:28   3:34 containerd -c /var/lib/rancher/k3s/agent/etc/containerd/config.toml -a /run/k3s/containerd/containerd.sock --state /run/k3s/containerd --root /var/lib/rancher/k3s/agent/containerd```
* I think I have containerd
* I can access crictl when I export the path to containerd socker `export CONTAINER_RUNTIME_ENDPOINT=unix:///run/k3s/containerd/containerd.socket`
* I can't find documentation or examples how to interact with containerd..
* I installed buildah, but it fails to talk to the runtime, so I try to export `BUILDAH_RUNTIME` but I don't know how to tell it is containerd, this does not work ```export BUILDAH_RUNTIME=unix:///run/k3s/containerd/containerd.socket```

When running buildah, I tried with and without sudo, because only root has access to the whole of k3s `/var/lib/rancher/k3s`
When running with sudo, I include the `-E` to preserve the exported env.

I would like to use the runtime to build image, I do not want to build image on kubernetes cluster (because I have `docker build -t potato:v1 .`  muscle memory, and I prefer not to write another yaml to build an image)
My k3s is running on arm64 and some images simply do not exist for arm64.

**Comment by caroline-suse-rancher** [comment]
@egidijus I don't see a clear question here, or statement of a problem with K3s, so I'm going to convert this to a discussion. You may also have more success getting answers to your questions in our users Slack channel :) https://rancher-users.slack.com/archives/CGGQEHPPW
